great_expectations/cli/suite.py

```diff
@@ -116,15 +182,52 @@ def suite_new(ctx, expectation_suite, interactive, profile, batch_request, no_ju
         )
         sys.exit(1)
 
-    _suite_new_workflow(
-        context=context,
-        expectation_suite_name=expectation_suite,
-        interactive=interactive,
-        profile=profile,
-        no_jupyter=no_jupyter,
-        usage_event=usage_event_end,
-        batch_request=batch_request,
-    )
+    # TODO: AJB 20210420 add better warning if user passes --non-interactive but either --profile or --batch-request (for this and suite_edit...)
+
+    # If user has provided a flag determining their configuration, skip prompt.
+    if (interactive is not None) or (profile is True) or (batch_request is not None):
+        # Assume batch needed if user passes --profile
+        if profile is True and (interactive is False or interactive is None):
+            cli_message(
+                "<green>Entering interactive mode since you passed the --profile flag</green>"
+            )
+            interactive = True
+        # Assume batch needed if user passes --batch-request
+        elif (batch_request is not None) and (
+            interactive is False or interactive is None
+        ):
+            cli_message(
+                "<green>Entering interactive mode since you passed the --batch-request flag</green>"
+            )
+            interactive = True
+    else:
```

Comment:
@anthonyburdi I feel that this selection logic should be moved into a dedicated method (e.g., in the `toolkit` module).  For example, the `datasource` selector is not in the primary `suite.py` module...

Reply:
Personally, I would like to see only broadly applicable helper methods in the `toolkit` module and command specific helper methods in the file with the command logic or a separate command specific helper or toolkit e.g. suite_helper.py or suite_toolkit.py. I don't feel strongly enough to die on that hill but since we are in a time crunch I'm planning to leave as is and we can always refactor later.